keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,742 @@
 
-This workflow describes the automation supporting creating a new (greenfield) backend bucket. Accessing this bucket is covered in [Sharing COSI Created Buckets](#sharing-cosi-created-buckets). COSI determines that a request is for an existing bucket when `BucketRequest.bucketInstanceName` is specified.
+BucketSecret is the name of the secret that COSI will generate with credentials to access the bucket. The same secret name has to be set in the podSpec as a projected secret volume.
 
-Here is the workflow:
+```
+    BucketAccessRequest - bar1                                BucketAccessClass - bac1
+    |---------------------------------------|                 |----------------------------------|
+    | spec:                                 |                 | spec:                            |
+    |   bucketAccessClassName: bac1         |                 |   provisioner: s3.amazonaws.com  |
+    |   bucketRequestName: br1              |                 |   parameters:                    | 
+    |   credentialsSecretName: bucketcreds1 |                 |     key: value                   |
+    |---------------------------------------|                 |----------------------------------|
+```
 
-+ COSI central controller sees a new `BucketRequest` (BR).
-+ COSI gets the `BR.BucketClass` (directly or via the matching default) and verifies that the BR's namespace is allowed.
-+ If BR.bucketInstanceName is empty, COSI creates the associated `Bucket`, filling in fields from the `BucketClass` and BR. Else, we have a brownfield BR.
-+ The sidecar sees the new `Bucket` and gRPC calls the driver to create a backend bucket.
-+ COSI updates the BR and `Bucket` instance to reference each other.
+###### 2. COSI creates an intermediate BucketAccess object
 
-In addition, the cosi-node-adapter sees a new app pod references a BAR:
-+ cosi-node-adapter receives _NodeStageVolume_ request from kubelet.
-+ adapter add a finalizer to both the BAR.BA and the BAR.BR.
-+ the finalizer value is `pod.name`
+The ControllerManager creates a BucketAccess object by copying fields over from both BucketAccessRequest and BucketAccessClass. This step can only proceed if the BucketAccessClass and Bucket already exist. The Bucket's `BucketReady` condition should be true. This BucketAccess object is an intermediate object, with its condition `AccessGranted` set to false and credentialsSeret pointing to a `nil` value. This indicates that access credentials are yet to be generated in the OSP.
 
-## Sharing COSI Created Buckets
-This is the greenfield -> brownfield access use case, where COSI has created the `Bucket` instance and the driver has provisioned a new bucket. Now, we want to share access to this bucket in other namespaces.
+More information about BucketAccess is [here](#bucketaccess)
 
-![ShareBucket Workflow](images/share-bucket.png)
+```
+    Bucket Access - bar-$uuid
```

Comment:
Why is this not just status on BucketAccessRequest?  Is there really a need for access to be 2 objects?

Reply:
Yes, access transfers is not possible with namespace only resources.

Reply:
What is an access transfer?  Why can't you just make a new request?  I'm trying to see how little we can get away with here.